### PR TITLE
Drop `viper-config` config for tests and get rid of viper dependency

### PR DESCRIPTION
**Pull request by dims**
At one point support for `viper-config` was added, but we never actually used it anywhere in our CI jobs or test infra. This is a millstone on the neck as it drags in a whole bunch of dependencies into our vendor/ directory.

Let's please remove it (and drop 22k lines of code)

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
e2e.test: removed the  `--viper-config` flag.  If you were previously using this to pass flags to `e2e.test` via a file, you will need to pass them directly on the command line, e.g. `e2e.test --e2e-output-dir`
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
:heart:

File: go.mod
Comment (aojea) [inline go.mod:410]: is good to downgrade?

Comment (dims) [reply go.mod:410]: it's a transitive dependency and we don't carry the code in our vendor/ directory, so we should be fine

```
[dims@dims-a02 11:46] ~/go/src/k8s.io/kubernetes ⟩ go mod graph | grep github.com/spf13/jwalterweatherman
github.com/spf13/viper@v1.7.0 github.com/spf13/jwalterweatherman@v1.0.0
[dims@dims-a02 11:46] ~/go/src/k8s.io/kubernetes ⟩ go mod graph | grep github.com/spf13/viper | grep -v "^github.com/spf13/viper@v1.7.0 "
github.com/spf13/cobra@v1.1.1 github.com/spf13/viper@v1.7.0
github.com/spf13/cobra@v1.0.0 github.com/spf13/viper@v1.7.0
github.com/spf13/cobra@v0.0.3 github.com/spf13/viper@v1.7.0
```

**Comment by liggitt** [comment]
I'm in favor of dropping the dependency... is this used here: https://github.com/kubernetes-sigs/cluster-api/blob/master/test/framework/kubetest/run.go#L141 ?

**Comment by dims** [comment]
@liggitt i'll get @vincepri and @fabriziopandini to peek that usage.

**Comment by liggitt** [comment]
dep drop lgtm, will approve once this has sig-testing ack

**Comment by dims** [comment]
Nice catch @liggitt looks like @randomvariable has one use in CI, slack traffic here:
https://kubernetes.slack.com/archives/C8TSNPY4T/p1622822716236300?thread_ts=1622822568.234800&cid=C8TSNPY4T

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/kubetest/conformance-fast.yaml

**Comment by randomvariable** [comment]
I'll try and remove from the CAP* consumers next week. Thanks for letting us know.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Comment by liggitt** [comment]
/approve
for dependency update

/hold in case you want to traffic direct this in, @dims

**Comment by spiffxp** [comment]
/kind deprecation
I edited the release note and I'm slapping this label on here to get this included in the "Deprecations" section of release notes.

I know this isn't setting a window before removal, but we have no guarantees regarding the "API" of `e2e.test`, and we haven't been able to find any clear usage of this flag in public.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
+1 with my SIG Testing Chair and SIG Architecture Conformance Subproject Owner hats on

The main consumer I was concerned about was sonobuoy, since that's what the vast majority of vendors use to submit results to cncf/k8s-conformance.  They don't use it, and I note they've also dropped viper over time.

**Comment by aojea** [comment]
> /hold in case you want to traffic direct this in, @dims

@dims :fire:

**Comment by dims** [comment]
@aojea waiting on https://github.com/kubernetes-sigs/cluster-api/issues/4729

**Comment by aojea** [comment]
ups, sorry, didn't see that :sweat_smile:

**Pull request merged**
